wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -421,7 +482,53 @@ protected RedisURI buildRedisURI(String hostname, int port) {
     return uriBuilder.build();
   }
 
+  /**
+   * Creates a ping connection factory that can be used by CacheMonitor.
+   * This factory encapsulates all the authentication and connection logic.
+   */
+  private CacheMonitor.PingConnectionFactory createPingConnectionFactory() {
+    return (hostname, port, isReadOnly) -> {
+      ClientResources resources = ClientResources.builder().build();
+      try {
+        RedisURI redisUri = buildRedisURI(hostname, port);
+        RedisClient client = RedisClient.create(resources, redisUri);
+        StatefulRedisConnection<byte[], byte[]> conn = client.connect(new ByteArrayCodec());
+
+        // Set READONLY mode for RO endpoint in cluster mode
+        if (isReadOnly) {
+          try {
+            conn.sync().readOnly();
+          } catch (RedisCommandExecutionException e) {
+            if (e.getMessage().contains("ERR This instance has cluster support disabled")) {
+              LOGGER.fine("Note: this cache cluster has cluster support disabled");
+            } else {
+              throw e;
+            }
+          }
+        }
+
+        return conn;
```

Comment:
This code looks very similar to the code in `createConnectionPool()`. 

Can you re-factor it out so you can re-use code from multiple places?